lib/vnet/unified_cluster_config_provider.go

```diff
@@ -0,0 +1,152 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package vnet
+
+import (
+	"context"
+	"slices"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/utils"
+)
+
+// UnifiedClusterConfig is a unified VNet configuration for all clusters
+// the user is logged in to.
+type UnifiedClusterConfig struct {
+	// ClusterNames contains the name of each root or leaf cluster the user is
+	// logged in to. SSH hosts are reachable via VNet SSH at subdomains of
+	// these cluster names.
+	ClusterNames []string
```

Comment:
should clusterNames, ProxyPublicAddrs, CustomDNSZones and IPv4CidrRanges have the same length or all of them can have different lenghts?

Reply:
ClusterNames and ProxyPublicAddrs will generally be the same length unless there are duplicates that get deduplicated.

Reply:
I was wondering if we should make them 

[]struct{ 
clustername string
proxypublicAddr string
...
}{

}

instead of sending different arrays and someone assuming they have the same size - i.e. cluster name and proxy public addrs - and cause a panic